File: data/gpt-4o/readable_format_HPEC/cpp/file204.cpp
#include <stdio.h>
#define DATA_TYPE double
void init_array(int n, DATA_TYPE p[], DATA_TYPE a[n][n]) {
 for (int i = 0; i < n; i++) {
 p[i] = 1.0 / n;
 for (int j = 0; j < n; j++) {
 a[j][i] = 1.0 / n;
 }
 }
}

File: data/llama3_8B_ins_results/readable_format_HPEC/fortran/file204.f90
subroutine init_array(n, p, a)
implicit none

DATA_TYPE, dimension(n, n) :: a
DATA_TYPE, dimension(n) :: p
integer :: n
integer :: i, j
do i = 1, n
p(i) = 1.0D0  / n
do j = 1, n
a(j, i) =  1.0D0 / n
end do
end do
end subroutine
